### DISCUSSION TITLE
How to combine `OneOrMany` with `HashMap`?

### DISCUSSION THREAD

**Discussion by mainrs** [jonasbb/serde_with#768] [Q&A] [answered]
I basically want this:

```rust
#[serde_as]
pub struct A {
    #[serde_as(as = "HashMap<String, OneOrMany<_>>")]
    pub keys: HashMap<String, Vec<OtherStruct>>,
}
```

I get an error:

```
the trait bound `HashMap<std::string::String, OneOrMany<Same>>: DeserializeAs<'_, _>` is not satisfied
   -->...
    |
72  | #[serde_as]
    | ^^^^^^^^^^^ the trait `DeserializeAs<'_, _>` is not implemented for `HashMap<std::string::String, OneOrMany<Same>>`

```

**Comment by jonasbb** [comment, marked as answer]
I think the issue is your use of `String` in the attribute. Try replacing it with a `_` like
`#[serde_as(as = "HashMap<_, OneOrMany<_>>")]`. Unfortunately if you want to use the normal serialization behavior of a type you have to use `_` since just writing the type will lead to the error you report.
